### ISSUE TITLE
Fix Subproject Paths for Commit

### ISSUE THREAD

**Issue by JoelProminic** [prominic/Moonshine-IDE#697]
A customer reported errors when cloning or comitting to a private repository.  I wasn't able to reproduce the cloneing issue on macOS or Windows, but I did notice this behavior with commit for a project with subprojects:
1. Clone https://github.com/prominic/Moonshine-IDE.git (you could also test this with another project with subprojects)
2. Open at least the ide/MoonshineDESKTOPevolved project
3. Make a cosmetic (i.e. whitespace) to build/build.xml
4. Try to commit build.xml with Git > Commit
5. BUG:  console shows this error:  `fatal: pathspec 'ide/MoonshineDESKTOPevolved/build/build.xml' did not match any files`

I confirmed this with the latest nightly build on macOS.  I was able to reproduce the error with an older build on Windows as well.

I can reproduce similar errors on the command line:

    $ git add ide/MoonshineDESKTOPevolved/build/build.xml
    fatal: pathspec 'ide/MoonshineDESKTOPevolved/build/build.xml' did not match any files
    $ git commit -m "Test commit" ide/MoonshineDESKTOPevolved/build/build.xml
    error: pathspec 'ide/MoonshineDESKTOPevolved/build/build.xml' did not match any file(s) known to git

This occurs because `git status --porcelain` returns the path relative to the local repository root.  Moonshine needs to either convert the paths to be relative to the project directory, or run them from the repository root.  The [git rev-parse](https://git-scm.com/docs/git-rev-parse) command may be useful for this.

    $ git rev-parse --show-toplevel
    /Users/joelanderson/Documents/Flex/temp_workspace/Moonshine-IDE_TestCommit
    $ git rev-parse --show-prefix
    ide/MoonshineDESKTOPevolved/

It looks like this behavior was already fixed for Reverting files from the Git > Status menu.

We can discuss which milestone this bug should be assigned to, or if we want to wait on this until we refactor the Git logic (#682).

**Comment by rat-moonshine** [comment]
Good you mentioned that it's working on Git-Status/Revert window, I notice we're running the commands from repository root (instead of sub-project directory), and we also have the information available. I don't remember now why we chose to do this in case of Status/Revert, but this also more later implementation than Commit, too.

I set the Commit command to run from the same repository root instead of sub-project directory, and that fix the problem. `git status --porcelain` used mainly for Commit and Status windows, I don't think any other place requires this immediate path-change; We can check them if we faced any similar problem from other sections, though.

I updated these changes to branch features/issue_637_git_merge.

**Comment by piotrzarzycki21** [comment]
Could your changes could be a part of master ?

**Comment by rat-moonshine** [comment]
That can be done. Once I confirm the handlings with Joel.

**Comment by rat-moonshine** [comment]
@piotrzarzycki21 Can we cherry-pick selected lines, instead of overwriting a complete file?

The commit-command class that is in #637 contains codes which we're still discussing on git-tuning issue. So I don't want a full-overwrite of the particular file. If cherry-picking is not possible against part of a file then I'll just copy the single-line fix to master.

**Comment by piotrzarzycki21** [comment]
> @piotrzarzycki21 Can we cherry-pick selected lines, instead of overwriting a complete file?
> 
> The commit-command class that is in #637 contains codes which we're still discussing on git-tuning issue. So I don't want a full-overwrite of the particular file in master immediately. If cherry-picking is not possible against part of a file then I'll just copy the single-line fix to master.

You can cherry pick file and commit as far as I know, but not part of the file. I really don't understand why you did changes on that branch  ? Could you explain to me. Even tests are not sufficient, cause when you push changes to master without all that code which is in #637 it will different environment and something will work differently.

**Comment by rat-moonshine** [comment]
As you're asking, maybe it could have been better if I tried the fix to master directly. Since many recent changes we were testing on that branch, maybe to keep things in testing mostly I go by the branch fix. 

Anyway, I'll copy the single line fix to master and give a test.

**Comment by rat-moonshine** [comment]
I applied the fix in master. Seems to be working well in my tests. 

@JoelProminic please, give a test at some time.

**Comment by JoelProminic** [comment]
I did some tests, and this seems to work properly on macOS and Windows.   

Type | Action | macOS | Windows
-----|--------|-------|--------
Single project | Commit | Works | Works
Single project | Revert | Works | Works
Subproject | Commit | Works | Works
Subproject | Revert | Works | Works
